src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1184,7 +1184,7 @@ public override BoundNode VisitLocalDeclaration(BoundLocalDeclaration node)
                 return null;
             }
 
-            bool inferredType = node.DeclaredType.InferredType;
+            bool inferredType = node.DeclaredTypeOpt?.InferredType ?? false; // a declaration with an inferred type will always store its DeclaredType
```

Comment:
>node.DeclaredTypeOpt?.InferredType ?? false; // a declaration with an inferred type will always store its DeclaredType [](start = 32, length = 118)

Is ```InferredType``` value the only reason for duplicating ```BoundTypeExpression``` for the ```var``` case? Please move the property to ```BoundLocalDeclaration``` and get rid of the duplication.

Reply:
A different type may be inferred for each local declaration, so duplication will pose that information.

Reply:
> A different type may be inferred for each local declaration, so duplication will pose that information.

How and where does that help? We still have only one syntax node for the ```var``` keyword. I am not suggesting to make any changes to how we infer the type of the local and what type do we capture in the LocalSymbol. 

---
In reply to: [255623589](https://github.com/dotnet/roslyn/pull/32843#discussion_r255623589) [](ancestors = 255623589)

Reply:
> How and where does that help? We still have only one syntax node for the `var` keyword. I am not suggesting to make any changes to how we infer the type of the local and what type do we capture in the LocalSymbol.

In a var case, for every bound local we do this:

```csharp
 if (isVar)
            {
                aliasOpt = null;
                bindType = true;

                initializerOpt = BindInferredVariableInitializer(diagnostics, value, valueKind, localSymbol.RefKind, declarator);

                // If we got a good result then swap the inferred type for the "var" 
                TypeSymbol initializerType = initializerOpt?.Type;
                if ((object)initializerType != null)
                {
                    declTypeOpt = TypeSymbolWithAnnotations.Create(initializerType);
```

We then use the `declTypeOpt` when creating the `boundDeclType` as follows:

```csharp
                boundDeclType = new BoundTypeExpression(typeSyntax, aliasOpt, inferredType: isVar, dimensionsOpt: invalidDimensions.ToImmutableAndFree(), type: declTypeOpt.TypeSymbol);
            }

            return new BoundLocalDeclaration(associatedSyntaxNode, localSymbol, boundDeclType, initializerOpt, arguments, hasErrors);
```

So every boundNode will have a different type.

Reply:
> So every boundNode will have a different type.

How and where does that help? I understand we were doing this before. The question is why do we have to keep doing this? We are changing things for other scenarios.

---
In reply to: [257821406](https://github.com/dotnet/roslyn/pull/32843#discussion_r257821406) [](ancestors = 257821406)